downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/FetchTestReactomeOntologyFiles.java

```diff
@@ -0,0 +1,127 @@
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.gk.persistence.MySQLAdaptor;
+
+public class FetchTestReactomeOntologyFiles {
+	private static final Logger logger = LogManager.getLogger();
+	private static Connection connect = null;
+	private static Statement statement = null;
+	private PreparedStatement preparedStatement = null;
+	private static ResultSet resultSet = null;
+	
+	public static void execute(MySQLAdaptor dba, String username, String password, String host, String database, String releaseNumber) throws SQLException, ClassNotFoundException, UnsupportedEncodingException, FileNotFoundException, IOException {
+		logger.info("Running FetchTestReactomeOntologyFiles...");
+		Class.forName("com.mysql.jdbc.Driver");
+		connect = DriverManager.getConnection("jdbc:mysql://" + host + "/" + database + "?" + "user=" + username + "&password=" + password);
+		statement = connect.createStatement();
+		resultSet = statement.executeQuery("SELECT ontology FROM Ontology");
+		// The returned value is a single blob composed of binary and text. The three files produced by this step (pprj, pins, pont) are found within this blob.
+		// A handful of regexes and conditional statements are used to handle this data and output the 3 files. 
+		
+		PrintWriter pprjWriter = new PrintWriter("reactome_data_model.pprj");
+		PrintWriter pontWriter = new PrintWriter("reactome_data_model.pont");
+		PrintWriter pinsWriter = new PrintWriter("reactome_data_model.pins");
+		
+		while (resultSet.next()) {
+			
+			Blob blob = resultSet.getBlob("ontology");
+			InputStream is = blob.getBinaryStream();
+			BufferedReader br = new BufferedReader(new InputStreamReader(is));
+
+			String dateTime = "";
+			int dateTimeCounter = 0;
+			boolean pprjSwitch = true;
+			boolean pontSwitch = false;
+			String str;
+			logger.info("Generating reactome_data_model.pprj, reactome_data_model.pont, and reactome_data_model.pins files...");
+			while ((str = br.readLine()) != null) {
+				
+				String[] splitLine = str.split(";");
+				if (splitLine.length > 1) {
+					// A very specific regex for matching a datetime string -- Could probably be matched to a specific format that looks cleaner
+					if (splitLine[1].matches("( [A-Z][a-z]{2}){2} [0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2} [A-Z]{3} [0-9]{4}")) {
+						dateTime = ";" + splitLine[1] + "\n";
+						dateTimeCounter++;
+						if (pprjSwitch) {
+							Files.write(Paths.get("reactome_data_model.pprj"), dateTime.getBytes(), StandardOpenOption.APPEND);
+						}
+						if (dateTimeCounter == 2) {
+							Files.write(Paths.get("reactome_data_model.pont"), dateTime.getBytes(), StandardOpenOption.APPEND);
+						}
+						if (dateTimeCounter == 3) {
+							Files.write(Paths.get("reactome_data_model.pins"), dateTime.getBytes(), StandardOpenOption.APPEND);
+						}
+						continue;
+					}
+				}
+				str += "\n";
```

Comment:
It's unusual to re-assign to a variable holding the value of iteration in a loop.  A new variable might be better.

Reply:
For this module in particular, I had to channel a bit of my inner Perlness to get things to cooperate. I can look into cleaning it up so that it follows good practices though

Reply:
Be one with the shebang line :smile:

Reply:
Created a String variable, 'line' that is used throughout the code block instead of the 'str' one created by the readLine call